test/specs/rules/require-matching-function-name.js

```diff
@@ -45,10 +65,30 @@ describe('rules/require-matching-function-name', function() {
             assertNoErrors('var test = {foo: function() {}};');
         });
 
+        it('should NOT report function name mismatch when assigning anonymous to property', function() {
+            assertNoErrors('var test = {foo: function() {}};');
+        });
+
         it('should NOT report function name mismatch when property name is reserved word', function() {
             assertNoErrors('var test = {delete: function $delete() {}};');
         });
 
+        it('should NOT report function name mismatch when property name is dynamic', function() {
+            assertNoErrors('var test = {["x" + 2]: function bar(){}};');
+        });
+
+        it('should NOT report function assignation to an expression', function() {
+            assertNoErrors('var test = {}; test["x" + 2] = function bar(){};');
+        });
+
+        it('should NOT report function name match on ES6 destructuring', function() {
+            assertNoErrors('let [ bar ] = [ function bar(){} ];');
+        });
+
+        it.skip('should report function name mismatch on ES6 destructuring', function() {
```

Comment:
For a second I though we need to check names for ES6-destructuring. Feels like need to drop this test.